lib/utils/ui/user_avatar/dynamic_user_avatar.dart

```diff
@@ -0,0 +1,41 @@
+import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/utils/ui/user_avatar/user_avatar.dart";
+import "package:proxima/viewmodels/dynamic_user_avatar_view_model.dart";
+
+/// This widget display a user's avatar.
+class DynamicUserAvatar extends HookConsumerWidget {
+  static const _loadingDisplayName = "";
+
+  final UserIdFirestore? uid;
+  final double radius;
+  final VoidCallback? onTap;
+
+  const DynamicUserAvatar({
+    super.key,
+    required this.uid,
+    required this.radius,
+    this.onTap,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final displayNameAsyncValue =
+        ref.watch(dynamicUserAvatarViewModelProvider(uid));
+
+    return displayNameAsyncValue.when(
+      data: (displayName) => UserAvatar(
+        displayName: displayName,
+        radius: radius,
+        onTap: onTap,
+      ),
+      loading: () => UserAvatar(
```

Comment:
Will `UserAvatar` ever be used outside of the `DynamicUserAvatar`? If not, I think it would make sense to combine the two as one class, as they have the parameters and structures. 
After combining, you could assign the `CircleAvatar` widget used it `UserAvatar` to a variable, to maintain the code readability.

Let me know what you think :)

Reply:
Yes, the idea is to keep the `UserAvatar` for parts of the code that already have the necessary information to display the widget. On the other side, the `DynamicUserAvatar` widget is useful when the necessary information is not given.

For example, for the `PostCard` widget. Our model for posts already provides the `displayName`, so using the `DynamicUserAvatar` would fetch twice the `displayName` of the user.

Reply:
Okay! Maybe we can call it `StaticUserAvatar` then?